Fix -Wreturn-mismatch errors wth gcc 14.

Also fix two print format warnings.

```
In file included from /usr/include/glib-2.0/glib.h:64:
imap_notify.c: In function 'check_new_debounced':
/usr/include/glib-2.0/glib/gmessages.h:671:16: error: 'return' with a value, in
function returning void [-Wreturn-mismatch]
  671 |         return (val); \
      |                ^
imap_notify.c:372:9: note: in expansion of macro 'g_return_val_if_fail'
  372 |         g_return_val_if_fail(item != NULL, NULL);
      |         ^~~~~~~~~~~~~~~~~~~~
imap_notify.c:370:13: note: declared here
  370 | static void check_new_debounced(FolderItem *item)
      |             ^~~~~~~~~~~~~~~~~~~
```

--- sylpheed-imap-notify-1.1.0-r1/imap_notify.c
+++ sylpheed-imap-notify-1.1.0-r2/imap_notify.c
@@ -369,7 +369,7 @@
 
 static void check_new_debounced(FolderItem *item)
 {
-	g_return_val_if_fail(item != NULL, NULL);
+	g_return_if_fail(item != NULL);
 
 	if (item->cache_dirty) return;
 	item->cache_dirty = TRUE;
@@ -400,7 +400,7 @@
 
 	debug_print("IMAP mailbox status:"
 			"\"%s\" messages: %d recent: %d unseen: %d "
-			"uid_next: %zu uid_validity: %zu\n", str,
+			"uid_next: %u uid_validity: %u\n", str,
 			status.has_messages ? status.messages : -1,
 			status.has_recent ? status.recent : -1,
 			status.has_unseen ? status.unseen : -1,
@@ -645,7 +645,7 @@
 	Folder *folder = NULL;
 	IMAPNotifySession *notify_session;
 
-	g_return_val_if_fail(session != NULL, NULL);
+	g_return_if_fail(session != NULL);
 
 	for (cur = folder_get_list(); cur != NULL; cur = cur->next) {
 		folder = cur->data;
@@ -708,8 +708,8 @@
 	IMAPSession *imap_session;
 	IMAPNotifySession *imap_notify_session;
 
-	g_return_val_if_fail(account != NULL, NULL);
-	g_return_val_if_fail(account->folder != NULL, NULL);
+	g_return_if_fail(account != NULL);
+	g_return_if_fail(account->folder != NULL);
 
 	/* No NOTIFY session yet. Try to steal one from the folder */
 
